src/main/kotlin/snyk/common/SnykCachedResults.kt

```diff
@@ -38,68 +49,126 @@ class SnykCachedResults(val project: Project) {
+                        }
+                }
+
+                override fun scanningIacError(snykError: SnykError) {
+                    currentIacResult = null
+                    currentIacError =
+                        when {
+                            snykError.message.startsWith(SnykToolWindowPanel.NO_IAC_FILES) -> null
+                            snykError.message.startsWith(SnykToolWindowPanel.AUTH_FAILED_TEXT) -> null
+                            else -> snykError
+                        }
+                }
+
+                override fun scanningContainerError(snykError: SnykError) {
+                    currentContainerResult = null
+                    currentContainerError =
+                        when {
+                            snykError == ContainerService.NO_IMAGES_TO_SCAN_ERROR -> null
+                            snykError.message.startsWith(SnykToolWindowPanel.AUTH_FAILED_TEXT) -> null
+                            else -> snykError
+                        }
+                }
+
+                override fun scanningSnykCodeError(snykError: SnykError) {
+                    AnalysisData.instance.resetCachesAndTasks(project)
+                    currentSnykCodeError = snykError
+                }
             }
-
-            override fun scanningOssError(snykError: SnykError) {
-                currentOssResults = null
-                currentOssError =
-                    when {
-                        snykError.message.startsWith(SnykToolWindowPanel.NO_OSS_FILES) -> null
-                        snykError.message.startsWith(SnykToolWindowPanel.AUTH_FAILED_TEXT) -> null
-                        else -> snykError
-                    }
-            }
-
-            override fun scanningIacError(snykError: SnykError) {
-                currentIacResult = null
-                currentIacError =
-                    when {
-                        snykError.message.startsWith(SnykToolWindowPanel.NO_IAC_FILES) -> null
-                        snykError.message.startsWith(SnykToolWindowPanel.AUTH_FAILED_TEXT) -> null
-                        else -> snykError
-                    }
-            }
-
-            override fun scanningContainerError(snykError: SnykError) {
-                currentContainerResult = null
-                currentContainerError =
-                    when {
-                        snykError == ContainerService.NO_IMAGES_TO_SCAN_ERROR -> null
-                        snykError.message.startsWith(SnykToolWindowPanel.AUTH_FAILED_TEXT) -> null
-                        else -> snykError
-                    }
+        )
+
+        project.messageBus.connect().subscribe(
+            SnykCodeScanListenerLS.SNYK_SCAN_TOPIC,
+            object : SnykCodeScanListenerLS {
+                val logger = logger<SnykCachedResults>()
+                override fun scanningStarted(snykScan: SnykScanParams) {
+                    logger.info("scanningStarted for project ${project.name}, emptying cache.")
```

Comment:
Should we not actually clear the cache here? It's just logging that it is.

Reply:
wrong log message :)